plugins/tech-insights-backend/src/service/router.ts

```diff
@@ -103,16 +103,30 @@ export async function createRouter<
         checks,
         entities,
       }: { checks: string[]; entities: CompoundEntityRef[] } = req.body;
+      const errors: string[] = [];
       const tasks = entities.map(async entity => {
         const entityTriplet =
           typeof entity === 'string' ? entity : stringifyEntityRef(entity);
-        const results = await factChecker.runChecks(entityTriplet, checks);
-        return {
-          entity: entityTriplet,
-          results,
-        };
+        try {
+          const results = await factChecker.runChecks(entityTriplet, checks);
+          return {
+            entity: entityTriplet,
+            results,
+          };
+        } catch (e: any) {
+          const errorMessage = `Failed to run check for entity ${entityTriplet} due to error: ${e.message}`;
+          logger.error(errorMessage);
+          errors.push(errorMessage);
+          return {
+            entity: entityTriplet,
+            results: [],
+          };
+        }
       });
       const results = await Promise.all(tasks);
+      if (errors.length > 0) {
+        return res.json({ errors, results });
```

Comment:
So if the result was entirely successful the response shape is an array, but if anything fails it's an object? It's a bit of a tricky contract. Just checking that this is intentional.

Would it be more useful to instead put the error in each entry? making it

```ts
return {
  entity: entityTriplet,
  results: [],
  error: <error>
};
```

instead.

Reply:
Ah good spot yes that makes sense